pipeline/authn/authenticator_cookie_session.go

```diff
@@ -155,6 +155,10 @@ func forwardRequestToSessionStore(r *http.Request, checkSessionURL string, prese
 
 	// We need to make a COPY of the header, not modify r.Header!
 	for k, v := range r.Header {
+		// remove Accept-Encoding to let the transport handle gzip
+		if k == "Accept-Encoding" {
```

Comment:
This took us a bit to figure out — it seems that when this is set, it turns off automatic decompressing. So omitting it, lets it do what it needs to do. (also h/t @rwos for finding the associated ticket and helping me fix this)

Reply:
@aeneasr just fyi, I also added a test the header is stripped (and stays stripped). :)

Reply:
Nice find! I assume this is similar to https://github.com/ory/oathkeeper/pull/952#discussion_r850728149 ?